src/Test/AbstractIntegrationTestCase.php

```diff
@@ -269,14 +262,27 @@ function (FixerInterface $fixer) {
                 new NullCacheManager()
             );
 
+            Tokens::clearCache();
             $runner->fix();
             $fixedInputCodeWithReversedFixers = file_get_contents($tmpFile);
 
-            $this->assertNotSame(
-                $fixedInputCode,
-                $fixedInputCodeWithReversedFixers,
-                sprintf('Set priorities must be significant. If fixers used in reverse order return same output then the integration test is not sufficient or the priority relation between used fixers should not be set. In "%s".', $case->getFileName())
-            );
+            // If output is different depends on rules order - we need to verify that the rules are ordered by priority.
+            // If not, any order is valid.
+            if ($fixedInputCode !== $fixedInputCodeWithReversedFixers) {
+                $this->assertGreaterThan(
+                    1,
+                    count(array_unique(array_map(
+                        function (FixerInterface $fixer) {
+                            return $fixer->getPriority();
+                        },
+                        $fixers
+                    ))),
+                    sprintf(
+                        'Rules priorities are not differential enough. If rules would be used in reverse order then final output would be different than the expected one. For that, different priorities must be set up for used rules to ensure stable order of them. In "%s".',
+                        $case->getFileName()
+                    )
+                );
+            }
```

Comment:
> I thought we were doing those kind of tests when priorities are different (or maybe I was in a specific use case).

We do unit tests for every rule separately.
We do integration tests for multiple rules combined if we think (or if it was proven) that they are conflicting one another. Sometimes they do, sometimes they dont. Then, this extra check will check that if they are conflicting indeed, the execution order is stable (so priorities are different).

-----------------------

> What I said earlier was: since we are running the fixer twice (in the right and reversed order) and we know that fixers without the same priority will generate diff, why don't we use this second run to check that two fixers with the same priority generate the same fix?

The purpose of the check is not "what should be the priority" but "what should be the output".
For that, our main concern is to get proper output - setting priorities are not the goal, but a way to achieve it.
If the outcome is the same regardless of order - it's good and we don't need to do anything else.
If the outcome is different for different orders - we need to check that priorities are set so we will always use same order.

-----------------------

> I've also noticed (thanks to your PR) that if fixers are specified in an order different from their priorities, the IntegrationTest will produce wrong fix.

There are two phases of itest:
- test in main order
- test in reverse order

main order is an order in which the Fixer tool itself will execute the rules - and they are sorted by https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/src/Test/AbstractIntegrationTestCase.php#L312 , not but order of json config in itest file.